controllers/component_build_controller_pac.go

```diff
@@ -535,7 +536,26 @@ func (r *ComponentBuildReconciler) UnconfigureRepositoryForPaC(ctx context.Conte
 
 	isAppUsed := IsPaCApplicationConfigured(gitProvider, pacConfig)
 	if !isAppUsed {
-		if webhookTargetUrl != "" {
+		componentList := &appstudiov1alpha1.ComponentList{}
+		if err := r.Client.List(ctx, componentList, &client.ListOptions{Namespace: component.Namespace}); err != nil {
+			log.Error(err, "failed to list components")
+			return "", "", "", err
+		}
+		gitUrl := strings.TrimSuffix(strings.TrimSuffix(component.Spec.Source.GitSource.URL, ".git"), "/")
+
+		sameRepoUsed := false
+		for _, comp := range componentList.Items {
+			if comp.Name == component.Name {
+				continue
+			}
+			componentUrl := strings.TrimSuffix(strings.TrimSuffix(comp.Spec.Source.GitSource.URL, ".git"), "/")
+			if componentUrl == gitUrl {
+				sameRepoUsed = true
+				break
+			}
+		}
+
+		if webhookTargetUrl != "" && !sameRepoUsed {
```

Comment:
It doesn't make sense to determine `sameRepoUsed` if `webhookTargetUrl` is empty string.

Reply:
we delete webhook ONLY if
webhookurl isn't empty (so we have some webhook)
AND we don't have other component with the same URL

Reply:
fixed